src/toil/jobStores/fileJobStore.py

```diff
@@ -144,8 +149,44 @@ def jobs(self):
                         pass  
 
     ##########################################
-    #Functions that deal with temporary files associated with jobs
-    ##########################################    
+    # Functions that deal with temporary files associated with jobs
+    ##########################################
+
+    def _importFile(self, otherCls, url):
+        if issubclass(otherCls, type(self)):
+            fd, absPath = self._getTempFile()
+            shutil.copyfile(self._extractPathFromUrl(url), absPath)
+            os.close(fd)
+            return self._getRelativePath(absPath)
+        else:
+            return super(FileJobStore, self)._importFile(otherCls, url)
+
+    def _exportFile(self, otherCls, jobStoreFileID, url):
+        if issubclass(otherCls, type(self)):
+            shutil.copyfile(self._getAbsPath(jobStoreFileID), self._extractPathFromUrl(url))
+        else:
+            super(FileJobStore, self)._exportFile(otherCls, jobStoreFileID, url)
+
+    @staticmethod
+    def readFromUrl(url, writable):
+        with open(FileJobStore._extractPathFromUrl(url), 'r') as f:
+            writable.write(f.read())
+
+    @staticmethod
+    def writeToUrl(readable, url):
+        with open(FileJobStore._extractPathFromUrl(url), 'w') as f:
+            f.write(readable.read())
+
+    @staticmethod
+    def _extractPathFromUrl(url):
+        """
+        :return: local file path of file pointed at by the given URL
+        """
+        return url.netloc + url.path
```

Comment:
That can't be right ...

`file:///foo/bar.txt` is equivalent to `file://localhost/foo/bar.txt`

You should verify in `supportsUrl()` that netloc is `''` or `'localhost'`.

Note that `file://some-remote-host/foo/bar.txt` is in fact a valid file URL but not one that FileJobStore can support.

`file://foo/bar.txt` is not a valid URL but urlparse() parses it. You should catch that, too.